File: java/gradle-build-logic/java-plugins/src/main/java/software/onepiece/gradle/pluginshowto/MyJavaLibraryPlugin.java
package software.onepiece.gradle.pluginshowto;

import org.gradle.api.NonNullApi;
import org.gradle.api.Plugin;
import org.gradle.api.Project;
import org.gradle.api.plugins.JavaLibraryPlugin;

@NonNullApi
abstract public class MyJavaLibraryPlugin implements Plugin<Project> {

    @Override
    public void apply(Project project) {
        project.getPlugins().apply(MyJavaBasePlugin.class);
        project.getPlugins().apply(JavaLibraryPlugin.class);
    }
}

File: clojure/gradle-build-logic/java-plugins/src/main/clojure/software/onepiece/gradle/pluginshowto/MyJavaLibraryPlugin.clj
(ns software.onepiece.gradle.pluginshowto.MyJavaLibraryPlugin
    (:import
      (org.gradle.api Project Plugin)
      (org.gradle.api.plugins JavaLibraryPlugin)
    )
    (:gen-class
      :name       software.onepiece.gradle.pluginshowto.MyJavaLibraryPlugin
      :implements [org.gradle.api.Plugin]
    )
  )

(defn -apply [this ^Project project]
  (do
  )
)

